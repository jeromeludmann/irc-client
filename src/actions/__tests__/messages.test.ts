import {
  messagesReceived,
  sendPart,
  sendPingToServer,
  sendPongToServer,
  sendQuit,
  sendUser,
} from "@app/actions/messages";

describe("messages actions", () => {
  it("should receive PRIVMSG", () => {
    expect(
      messagesReceived["PRIVMSG"](
        "server1",
        { nick: "nick1", user: "user", host: "host" },
        ["nick2", "hello world"],
      ),
    ).toMatchSnapshot();
  });

  describe("should send PART", () => {
    it("without message", () => {
      expect(sendPart("server1", "#channel")).toMatchSnapshot();
    });

    it("with message", () => {
      expect(sendPart("server1", "#channel", "Goodbye!")).toMatchSnapshot();
    });
  });

  it("should send PING", () => {
    expect(sendPingToServer("server1", "key")).toMatchSnapshot();
  });

  it("should send PONG", () => {
    expect(sendPongToServer("server1", "key")).toMatchSnapshot();
  });

  describe("should send QUIT", () => {
    it("without message", () => {
      expect(sendQuit("server1")).toMatchSnapshot();
    });

    it("with message", () => {
      expect(sendQuit("server1", "Goodbye!")).toMatchSnapshot();
    });
  });

  it("should send USER", () => {
    expect(sendUser("server1", "username", "Real Name")).toMatchSnapshot();
  });
});
