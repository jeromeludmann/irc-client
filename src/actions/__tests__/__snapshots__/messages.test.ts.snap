// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages should receive PRIVMSG 1`] = `
Object {
  "payload": Object {
    "text": "hello world",
    "user": Object {
      "host": "host",
      "nick": "nick1",
      "user": "user",
    },
  },
  "route": Object {
    "channelKey": "nick1",
    "serverKey": "server1",
  },
  "type": "MESSAGE/PRIVMSG",
}
`;

exports[`messages should send PART with message 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "PART #channel Goodbye!",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send PART without message 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "PART #channel ",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send PING 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "PING key",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send PONG 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "key": "key",
    },
    "route": Object {
      "channelKey": "@status",
      "serverKey": "server1",
    },
    "type": "MESSAGE/SEND_PONG_TO_SERVER",
  },
  "payload": Object {
    "raw": "PONG key",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send QUIT with message 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "QUIT Goodbye!",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send QUIT without message 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "QUIT ",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`messages should send USER 1`] = `
Object {
  "embeddedAction": undefined,
  "payload": Object {
    "raw": "USER username 0 * :Real Name",
  },
  "route": Object {
    "channelKey": "@none",
    "serverKey": "server1",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;
